flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/InternalTimersSnapshotReaderWriters.java

```diff
@@ -96,7 +96,7 @@ public AbstractInternalTimersSnapshotWriter(InternalTimersSnapshot<K, N> timersS
 		public final void writeTimersSnapshot(DataOutputView out) throws IOException {
 			writeKeyAndNamespaceSerializers(out);
 
-			InternalTimer.TimerSerializer<K, N> timerSerializer = new InternalTimer.TimerSerializer<>(
+			TimerHeapInternalTimer.TimerSerializer<K, N> timerSerializer = new TimerHeapInternalTimer.TimerSerializer<>(
```

Comment:
Isn't this a bit problematic if we add a new `TimerService` implementation other than the `HeapInternalTimerService`? The `InternalTimersSnapshot` is independent of the underlying timer service implementation and so should the timer serializer be.

Reply:
I think this is ok for the moment and we can adjust in case there will ever be another implementation of `InternalTimer `